client.go

```diff
@@ -0,0 +1,75 @@
+package nomadclient
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"errors"
+	dphttp "github.com/ONSdigital/dp-net/http"
+	"io/ioutil"
+	"net/http"
+	"strings"
+)
+
+type Nomad struct {
+	Client dphttp.Clienter
+	URL string
+	// Name needed for checker later.
+	// Name string
+}
+
+const prefix = "https://"
+var caCertPool *x509.CertPool
+
+func NewClient(nomadEndpoint, nomadCACert string, nomadTLSSkipVerify bool) (*Nomad, error){
+
+	dpHTTPClient := *dphttp.DefaultClient
+
+	if strings.HasPrefix(nomadEndpoint, prefix) {
+		tlsConfig, err := createTlsConfig(nomadCACert, nomadTLSSkipVerify)
+		if err != nil {
+			return nil, err
+		}
+
+		dpHTTPClient.HTTPClient.Transport = &http.Transport{TLSClientConfig: tlsConfig}
+	}
+	return  &Nomad{
+		Client: &dpHTTPClient,
+		URL:    nomadEndpoint,
+		// Name:   name,
+	}, nil
+}
+
+func createTlsConfig (nomadCACert string, nomadLSSkipVerify bool) (*tls.Config, error) {
+
+	if nomadCACert == "" && !nomadLSSkipVerify {
+		return nil, errors.New("invalid configuration with https but no CA cert or skip verification enabled")
+	}
+	if nomadCACert != "" {
```

Comment:
Probably don't need this if statement anymore as you have checked earlier if nomadCACert is empty and if it is returned (line 44). If the code gets to this point then it will never be empty.